polus-imagej-plugins/polus-imagej-image-integral-plugin/src/main.py

```diff
@@ -0,0 +1,164 @@
+    
+        # switch to images folder if present
+        if _in1.joinpath('images').is_dir():
+            _in1 = _in1.joinpath('images').absolute()
+
+        args.append([f for f in _in1.iterdir() if f.is_file()])
+        arg_len = len(args[-1])
+    else:
+        arg_types.append(None)
+        args.append([None])
+    
+    for i in range(len(args)):
+        if len(args[i]) == 1:
+            args[i] = args[i] * arg_len
+            
+    """ Set up the output """
+    out_types = { 
+        "DefaultIntegralImg": "RandomAccessibleInterval",
+        "SquareIntegralImg": "RandomAccessibleInterval",
+        "WrappedIntegralImg": "RandomAccessibleInterval",
+    }
+
+    """ Run the plugin """
+    try:
+        for ind, (in1_path,) in enumerate(zip(*args)):
+            if in1_path != None:
+
+                # Load the first plane of image in in1 collection
+                logger.info('Processing image: {}'.format(in1_path))
+                in1_br = BioReader(in1_path)
+                
+                # Convert to appropriate numpy array
+                in1 = ij_converter.to_java(np.squeeze(in1_br[:,:,0:1,0,0]),in1_type)
+                metadata = in1_br.metadata
+                fname = in1_path.name
+                dtype = ij.py.dtype(in1)
+            logger.info('Running op...')
+            if _opName == "DefaultIntegralImg":
+                out = ij.op().image().integral(in1)
+            elif _opName == "SquareIntegralImg":
+                out = ij.op().image().integral(in1)
+            elif _opName == "WrappedIntegralImg":
+                out = ij.op().image().integral(in1)
+            
+            logger.info('Completed op!')
+            if in1_path != None:
+                in1_br.close()
+            
+            # Saving output file to out
+            logger.info('Saving...')
+            out_array = ij_converter.from_java(out,out_types[_opName])
+            bw = BioWriter(_out.joinpath(fname),metadata=metadata)
+            bw.Z = 1
+            bw.dtype = out_array.dtype
+            bw[:] = out_array.astype(bw.dtype)
+            bw.close()
+            
+    except:
+        logger.error('There was an error, shutting down jvm before raising...')
+        raise
+            
+    finally:
+        # Exit the program
+        logger.info('Shutting down jvm...')
+        del ij
+        jpype.shutdownJVM()
+        logger.info('Complete!')
+
+if __name__=="__main__":
+
+    ''' Setup Command Line Arguments '''
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='DefaultIntegralImg,SquareIntegralImg,WrappedIntegralImg')
+    
+    # Add command-line argument for each of the input arguments
+    parser.add_argument('--opName', dest='opName', type=str,
+                        help='Operation to perform', required=False)
+    parser.add_argument('--in1', dest='in1', type=str,
+                        help='in1', required=False)
```

Comment:
Different ops have different input requirements. Rather than specifying which inputs are required here, I do checks in the code to handle. Some ops handled by the same plugin may have no overlap in input or output variable names.

Functionally, on the Python side, input/output paths are basically the same thing. I keep them separate for the purposes of properly casting data to the appropriate types in Java.